pgjdbc/src/main/java/org/postgresql/jdbc/TimestampUtils.java

```diff
@@ -463,25 +464,45 @@ public LocalDateTime toLocalDateTime(String s) throws SQLException {
 
   public synchronized Time toTime(Calendar cal, String s) throws SQLException {
     // 1) Parse backend string
-    Timestamp timestamp = toTimestamp(cal, s);
-
-    if (timestamp == null) {
+    if (s == null) {
       return null;
     }
-
-    long millis = timestamp.getTime();
-    // infinity cannot be represented as Time
-    // so there's not much we can do here.
-    if (millis <= PGStatement.DATE_NEGATIVE_INFINITY
-        || millis >= PGStatement.DATE_POSITIVE_INFINITY) {
-      throw new PSQLException(
-          GT.tr("Infinite value found for timestamp/date. This cannot be represented as time."),
-          PSQLState.DATETIME_OVERFLOW);
+    ParsedTimestamp ts = parseBackendTimestamp(s);
```

Comment:
this seems to be missing the handling of infinity and -infinity as unsupported by getTime